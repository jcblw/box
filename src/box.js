/** @jsx jsx */
import { jsx } from '@emotion/core'
import React from 'react'
import { propsToStyles } from './lib/helpers'
import { removeKeys } from './lib/remove-keys'
import * as flexStyles from './styles/flex'
import { cssToStyle } from './styles/helpers'
import * as spacingStyles from './styles/spacing'
import * as utilStyles from './styles/utils'

export const styleGuide = [flexStyles, spacingStyles, utilStyles]

export const Box = React.forwardRef((props, ref) => {
  const { Component, getStyles, css, styles, name } = props
  const customStyles = cssToStyle(css)
  const results = getStyles({ ...props })
  const otherProps = removeKeys(
    props,
    ...results.used,
    'Component',
    'getStyles',
    'css',
    'styles',
    'name'
  )
  if (name && styles) {
    results.styles.push({ styles, name })
  }
  if (customStyles) {
    results.styles.push(customStyles)
  }
  return <Component ref={ref} css={results.styles} {...otherProps} />
})

Box.displayName = 'Box'

Box.defaultProps = { Component: 'div', getStyles: propsToStyles(styleGuide) }
