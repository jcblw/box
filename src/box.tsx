/** @jsx jsx */
import { jsx } from '@emotion/core'
import React from 'react'
import { propsToStyles, cssToStyle } from './lib/helpers'
import { omitKeys } from './lib/omit-keys'
// import * as spacingStyles from './styles/spacing'
// import * as utilStyles from './styles/utils'
import { BoxProps } from './types'
import { useStyleGuideBox } from './style-guide'

export const Box = React.forwardRef<unknown, BoxProps>((props, ref) => {
  const { defaultComponent, styleGuide } = useStyleGuideBox()
  const getStyles = propsToStyles(styleGuide)
  const { Component = defaultComponent, css, styles, name } = props
  const customStyles = cssToStyle(css)
  const results = getStyles(props)
  const otherProps = omitKeys(
    props,
    'Component',
    'css',
    'styles',
    'name',
    ...results.used
  )
  if (name && styles) {
    results.styles.push({ styles, name })
  }
  if (customStyles) {
    results.styles.push(customStyles)
  }
  return <Component ref={ref} css={results.styles} {...otherProps} />
})

Box.displayName = 'Box'
