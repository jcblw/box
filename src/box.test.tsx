import React from 'react'
import Renderer from 'react-test-renderer'
import { Box } from './box'

test('Box should be present', () => {
  expect(Box).toBeTruthy()
})

test('Box should render properly', () => {
  const json = Renderer.create(<Box padding="m" />).toJSON()
  expect(json).toMatchSnapshot()
})
