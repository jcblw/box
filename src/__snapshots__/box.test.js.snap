// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box should render properly 1`] = `
<div
  className="css-19kzrtu"
/>
`;
